evalml/tests/component_tests/test_knn_imputer.py

```diff
@@ -0,0 +1,162 @@
+import numpy as np
+import pandas as pd
+import pytest
+from pandas.testing import assert_frame_equal
+from sklearn.impute import KNNImputer as Sk_KNNImputer
+
+from evalml.pipelines.components.transformers.imputers import KNNImputer
+
+
+@pytest.mark.parametrize("n_neighbors", [1, 2, 5])
+def test_knn_output(n_neighbors):
+    X = pd.DataFrame(
+        np.array(
+            [
+                [1, 1, 2],
+                [1, 1, np.nan],
+                [1, 1, 2],
+                [1, 1, 2],
+                [1, 2, 1],
+                [1, 1, 2],
+                [1, 1, 2],
+            ],
+        ),
+    )
+    sk_knn = Sk_KNNImputer(n_neighbors=n_neighbors)
+    knn = KNNImputer(n_neighbors)
+    assert_frame_equal(pd.DataFrame(sk_knn.fit_transform(X)), knn.fit_transform(X))
+
+
+@pytest.mark.parametrize("data_type", ["pd", "ww"])
+def test_knn_imputer_all_bool_return_original(data_type, make_data_type):
+    X = pd.DataFrame([True, True, False, True, True], dtype=bool)
+    y = pd.Series([1, 0, 0, 1, 0])
+    X = make_data_type(data_type, X)
+    y = make_data_type(data_type, y)
+    X_expected_arr = pd.DataFrame([True, True, False, True, True], dtype=bool)
+    imputer = KNNImputer()
+    imputer.fit(X, y)
+    X_t = imputer.transform(X)
+    assert_frame_equal(X_expected_arr, X_t)
+
+
+@pytest.mark.parametrize("data_type", ["pd", "ww"])
+def test_knn_imputer_boolean_dtype(data_type, make_data_type):
+    X = pd.DataFrame(
+        {
+            "some_nan": pd.Series([True, np.nan, False, np.nan, True], dtype="boolean"),
+        },
+    )
+    X.ww.init(logical_types={"some_nan": "BooleanNullable"})
```

Comment:
should we only init here based off if `data_type == ww`? we should either init or run `make_data_type` later.